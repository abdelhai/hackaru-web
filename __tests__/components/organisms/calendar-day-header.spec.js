import Factory from '@/__tests__/__setups__/factory';
import CalendarDayHeader from '@/components/organisms/calendar-day-header';
import { parse } from 'date-fns';
import enLocale from 'date-fns/locale/en';

describe('CalendarDayHeader', () => {
  let factory;
  let wrapper;

  beforeEach(() => {
    factory = new Factory(CalendarDayHeader, {
      propsData: {
        day: parse('2019-01-01T00:00:00'),
        locale: enLocale
      }
    });
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when click button', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.find('button').trigger('click');
    });

    it('emit click', () => {
      expect(wrapper.emitted('click')).toBeTruthy();
    });
  });
});
