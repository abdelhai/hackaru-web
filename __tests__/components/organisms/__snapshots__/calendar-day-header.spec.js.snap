// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarDayHeader render correctly 1`] = `
<header
  class=""
>
  <button>
    <h1>
      01
    </h1>
     
    <small>
      Tu
    </small>
  </button>
</header>
`;
